src/aks-preview/azext_aks_preview/_helpers.py

```diff
@@ -2,32 +2,164 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
+import errno
+import os
+import platform
 import re
+import stat
+import tempfile
+import yaml  # pylint: disable=import-error
 
 from azure.cli.command_modules.acs._helpers import map_azure_error_to_cli_error
 from azure.cli.core.azclierror import InvalidArgumentValueError, ResourceNotFoundError
 from azure.core.exceptions import AzureError
+from knack.log import get_logger
+from knack.prompting import NoTTYException, prompt_y_n
+from knack.util import CLIError
 
 from azext_aks_preview._client_factory import cf_nodepool_snapshots, cf_mc_snapshots
-from azext_aks_preview._consts import CONST_CONTAINER_NAME_MAX_LENGTH
 
+logger = get_logger(__name__)
 
-def _trim_fqdn_name_containing_hcp(normalized_fqdn: str) -> str:
-    """
-    Trims the storage blob name and takes everything prior to "-hcp-".
-    Currently it is displayed wrong: i.e. at time of creation cli has
-    following limitation:
-    https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/
-    error-storage-account-name
-
-    :param normalized_fqdn: storage blob name
-    :return: storage_name_without_hcp: Storage name without the hcp value
-    attached
+
+def which(binary):
+    path_var = os.getenv('PATH')
+    if platform.system() == 'Windows':
+        binary = binary + '.exe'
+        parts = path_var.split(';')
+    else:
+        parts = path_var.split(':')
+
+    for part in parts:
+        bin_path = os.path.join(part, binary)
+        if os.path.exists(bin_path) and os.path.isfile(bin_path) and os.access(bin_path, os.X_OK):
+            return bin_path
+
+    return None
+
+
+def print_or_merge_credentials(path, kubeconfig, overwrite_existing, context_name):
+    """Merge an unencrypted kubeconfig into the file at the specified path, or print it to
+    stdout if the path is "-".
     """
-    storage_name_without_hcp, _, _ = normalized_fqdn.partition('-hcp-')
-    if len(storage_name_without_hcp) > CONST_CONTAINER_NAME_MAX_LENGTH:
-        storage_name_without_hcp = storage_name_without_hcp[:CONST_CONTAINER_NAME_MAX_LENGTH]
-    return storage_name_without_hcp.rstrip('-')
+    # Special case for printing to stdout
+    if path == "-":
+        print(kubeconfig)
```

Comment:
It is recommended to use log instead of direct printing
Please refer to the [command_guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#general-patterns
) for more details

> Log to logger.error() or logger.warning() for user messages; do not use the print() function
Use the appropriate logging level for printing strings. e.g. logging.info(“Upload of myfile.txt successful”) NOT return “Upload successful”.